worldedit/manipulations.lua

```diff
@@ -38,6 +38,29 @@ function worldedit.set(pos1, pos2, node_names)
 	return worldedit.volume(pos1, pos2)
 end
 
+--- Sets param2 of a region.
+-- @param pos1
+-- @param pos2
+-- @param param2 Value of param2 to set
+-- @return The number of nodes set.
+function worldedit.set_param2(pos1, pos2, param2)
+	pos1, pos2 = worldedit.sort_pos(pos1, pos2)
+
+	local manip, area = mh.init(pos1, pos2)
+	local param2_data = {}
+
+	-- Set param2 for every node
+	for i in area:iterp(pos1, pos2) do
+		param2_data[i] = param2

```

Comment:
Please add a parameter to set_param2 to only change param2 if the node has paramtype2 "facedir" (or coloured facedir). Some mods use param2 for other purposes than rotating.

Reply:
I fail to see the use case here. The condition you mention seems kinda arbitrary.

Reply:
> Use case: Creating and saving tree schematics in MTG.

You'd also set param2 to air and leaves nodes although it should be 0.

Reply:
I am still not too happy with that idea. The main reason is that it is too specific. If I add a parameter for `facedir`, why not adding support for other `paramtype2`s as well? The end result will go to a lot of special cases, etc. and I don't know where this will end.

Better reserve this discussion for a different issue / PR?

Reply:
Then please add a note to the command's documentation.